### DISCUSSION TITLE
Further Ideas for Document Improvement

### DISCUSSION THREAD

**Discussion by f0rki** [f0rki/mapping-high-level-constructs-to-llvm-ir#34] [General]
Maybe we can start a discussion here on new ideas regarding this document!

**Comment by archfrog** [comment]
Yes, that would be awesome :-)

Personally, I can think of these off the top of my head:

1. A full "zero cost exception handling" example, this probably requires compiling some C++ programs into LLVM IR and see how its done.
2. Perhaps a chapter on lowering C++-style code to C-style code as LLVM IR is very low-level.  I can do this, it is rather simple.
3. Perhaps a chapter on handling Single Standard Assignment (SSA) form; LLVM IR allows for some tricks to avoid using this.

More will follow, everybody else reading this please feel free to add your comments :-)

**Reply by YurySolovyov** [reply]
Maybe say something about looping in regards to SSA or add a new page to control flow section.
Or, rename SSA page to looping and mention Phy nodes as a way to implement some looping patters (accumulator, and/or induction var?)
Or, some mix of the above :)

**Comment by ceymard** [comment]
* Garbage Collection / Reference counting (this could be a start)
* Aliasing ?
* Function attributes with a more detailed explanation than what the llvm doc covers ?

**Reply by YurySolovyov** [reply]
What I found confusing about LLVM docs about GC, is that they provide all these GC strategies, but they also say that LLVM does not actually implement them, and only gives some tools to *help* implementing GC.
It is not easy to understand how are these tools actually help.

**Reply by ceymard** [reply]
To be honest, I'm completely confused as well, which is why a guide about how this plays out would be a tremendous help for all of us wannabe language creators.

**Reply by f0rki** [reply]
GC seems a pretty big topic. I think it might be best to start by gathering helpful resources (i.e., Links, Books, Papers, Example Code, Links to Real-World Implementations etc.) on the topic of GC (and LLVM) in a new chapter.
